### ISSUE TITLE
BVA | Intake | Legacy opt-in | Form 10182 SOC Timeliness Update Required Within Caseflow

### ISSUE THREAD

**Issue by msj0nes** [department-of-veterans-affairs/caseflow#14206]
## Description

In order for a legacy issue to be opted-in to AMA, it must pass certain eligibility rules.  One rules is that a legacy appeal is eligible if the receipt date of the from is within 60 calendar days of the SOC or SSOC dates in VACOLS.  The Board of Veterans appeals has requested that Caseflow is inspected for the below as it relates to SOC-Opt In and some adjustments made to the calculations (see background/context).

This ticket incorporates an adjustment to delay the eligibility period by 1 day, and extend the eligibility period to end on the next weekday if it is scheduled to end on a weekend.

## Acceptance criteria
- [ ] When comparing the SOC/SSOC dates to the earliest eligible date, add 1 day to to the SOC/SSOC dates to start the period one day later
- [ ] If the SOC date or any SSOC dates + 1 day + 60 days falls on a Saturday or Sunday, expand the timing to include the following Monday (add 1-2 days)
- [ ] Add model tests
- [ ] Update documentation: [Legacy issue opt-in eligibility requirements](https://github.com/department-of-veterans-affairs/caseflow/wiki/Legacy-issue-opt-in-eligibility-requirements)

## Out of scope
- Handling if the last eligible day lands on a a federal holiday (see #14285 for handling holidays)

## Technical notes
The documentation linked above is useful for some additional context.  The code that needs to be updated is in `LegacyAppeal.soc_eligible_for_opt_in?`

## Background/context

**Timeliness Rules:** 
- SOCs are timely so long as the Form 10182 is received within one year of the date of notification of the rating decision even if more than 60 days from notification of the SOC. ✅(existing functionality)
>(_under 20.110(b) in order to opt into AMA_)
- Form 10182 should have been received within 60 calendar days of the SOC date in VACOLS
  - The first day of the period (i.e., the day on which the appealed decision was issued) should not be counted.  _For example, if the SOC is issued on April 30, 2020, the 60 day period should not begin to run until May 1, 2020_.  
  - If the 60th day falls on a Saturday, Sunday, or Federal holiday, the deadline should be extended until the next day which is not a Saturday, Sunday, or Federal holiday.  
  - The regulation does not exclude the counting of Saturdays, Sundays, or Federal holidays from the 60-day limit unless the time period expires on one of those days.”
  - When a holiday occurs on a Saturday, the Friday immediately before is the legal public holiday
  - When a holiday occurs on a Sunday, the Monday immediately after is the legal public holiday

**Board issue reported:**
https://appeals.cf.ds.va.gov/queue/appeals/7ee877cf-efb8-49fd-886e-e6384c6334d6
The SOC was issued on 12/17/19.  Exactly 60 calendar days from that date was 2/15/20, which was a Saturday.  Accordingly, the receipt of the Form 10182 on the following Monday, 2/17/20, is considered as the Form 10182 being timely received under 20.110(b) in order to opt into AMA.  However, Caseflow rejected our attempts to properly docket the issue as an opt-in from the SOC as being outside the window of eligibility.

**Comment by msj0nes** [comment]
##Approach
Tickets to be created and added to epic.
- Include weekends rule
- Update additional timeliness rule
- Distinguish error message
- Update remands

**Comment by leikkisa** [comment]
Hi @msj0nes - can you confirm whether these rules will impact all legacy issue opt-ins (for both BVA and VBA)?  I presume they're both the same and the answer is "Yes" but want to double check.

Just to make sure on this one:
> The first day of the period (i.e., the day on which the appealed decision was issued) should not be counted. For example, if the SOC is issued on April 30, 2020, the 60 day period should not begin to run until May 1, 2020.

So the eligible dates (for a non-weekend/holiday case) would be **May 1, 2020 - June 30, 2020** **including both May 1st and June 30th** ?.

**Comment by leikkisa** [comment]
@msj0nes 
The example provided above:
> Board issue reported:
https://appeals.cf.ds.va.gov/queue/appeals/7ee877cf-efb8-49fd-886e-e6384c6334d6
The SOC was issued on 12/17/19. Exactly 60 calendar days from that date was 2/15/20, which was a Saturday. Accordingly, the receipt of the Form 10182 on the following Monday, 2/17/20, is considered as the Form 10182 being timely received under 20.110(b) in order to opt into AMA. However, Caseflow rejected our attempts to properly docket the issue as an opt-in from the SOC as being outside the window of eligibility.

The receipt date is Mon, 17 Feb 2020.  However, I only see one legacy appeal in the review's legacy appeals - vacols_id 3268692.  It has an SOC date in VACOLS of Tue, 26 Apr 2016, and NOD date of Mon, 28 Dec 2015, and no SSOC dates.  Those are both well out of the window of eligibility, and there is another check that they cannot be before AMA launched on February 19, 2019.  Also on that appeal, the intake user selected that the veteran did NOT choose to opt-in their legacy appeals, so it would have been marked as ineligible even if the dates matched.

So there are many things going on, but nothing in Caseflow will help that legacy appeal become Caseflow unless the VACOLS data is changed.

**Comment by msj0nes** [comment]
### Asking BVA today the questions above 5/14/20

- Does the rules provided impact all legacy issue opt-ins (for both BVA and VBA)? 

- Using this rule. May 1st to June 30 would be exactly 60 days. To confirm, May 1st and June 30 should be included.

>The first day of the period (i.e., the day on which the appealed decision was issued) should not be counted. For example, if the SOC is issued on April 30, 2020, the 60 day period should not begin to run until May 1, 2020.

- Does the time zone we use for determining the current date matter? (to determine whether it's a weekend or not).

**Comment by msj0nes** [comment]
@Sjones352 @leikkisa response received from BVA

>From: Sanford, M
Sent: Tuesday, May 19, 2020 1:09 PM
To: Jones, M
Subject: RE: Caseflow Use of Calendar Days for Timeliness Checks [Now tracked via Github #14206
_________________________________

•	Does the rules provided impact all legacy issue opt-ins (for both BVA and VBA)?

**Yes, the rule impact all legacy issue opt-ins.**

•	Does the time zone we use for determining the current date matter? (to determine whether it's a weekend or not).

**I think the time zone should not matter.  There is no regulation governing which time zone is to be used.**

•	Using this rule. May 1st to June 30 would be exactly 60 days. To confirm, May 1st and June 30 should be included.
>The first day of the period (i.e., the day on which the appealed decision was issued) should not be counted. For example, if the SOC is issued on April 30, 2020, the 60 day period should not begin to run until May 1, 2020.

**In that example, May 1 would be included.  If the SOC was issued on April 30, 2020, the clock would begin to run on May 1 and the sixtieth day would fall on June 30.**
